test/unit/person_test.rb

```diff
@@ -1407,4 +1408,327 @@ def test_updated_not_changed_when_adding_notifiee_info
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    assert_equal [s1, s2, s3, s4].sort, person_to_keep.contributed_sops.sort
+    assert_equal [s2, s4, s5].sort, person_to_keep.created_sops.sort
+  end
+
+  test 'merge all types of resources' do
+    person_to_keep = FactoryBot.create(:person)
+    other_person = FactoryBot.create(:person)
+
+    types_without_creators = %w[Event Strain]
+    Person::RELATED_RESOURCE_TYPES.each do |resource_type|
+      sym = resource_type == 'SampleType' ? :simple_sample_type : resource_type.underscore.to_sym
+      FactoryBot.create(sym, contributor: other_person)
+      unless types_without_creators.include?(resource_type)
+        FactoryBot.create(sym, contributor: FactoryBot.create(:person), creator_ids: [other_person.id])
+      end
+    end
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    resource_types = Person::RELATED_RESOURCE_TYPES.length
+    added_isa_structure = 3 # An additional inv is created for the study, and inv and study are created for the assay
+    assert_equal resource_types + added_isa_structure, person_to_keep.contributed_items.length
+    assert_equal resource_types - types_without_creators.length, person_to_keep.created_items.length
+  end
+
+  test 'merge permissions without duplication' do
+    person_to_keep = FactoryBot.create(:person)
+    other_person = FactoryBot.create(:person)
+
+    sop_keep = FactoryBot.create(:sop)
+    sop_keep.policy.permissions.build(contributor: person_to_keep, access_type: Policy::MANAGING)
+    sop_keep.policy.save
+    sop_both = FactoryBot.create :sop
+    sop_both.policy.permissions.build(contributor: person_to_keep, access_type: Policy::MANAGING)
+    sop_both.policy.permissions.build(contributor: other_person, access_type: Policy::MANAGING)
+    sop_both.policy.save
+    sop_other = FactoryBot.create :sop
+    sop_other.policy.permissions.build(contributor: other_person, access_type: Policy::MANAGING)
+    sop_other.policy.save
+
+    assert sop_keep.can_manage?(person_to_keep)
+    assert sop_both.can_manage?(person_to_keep)
+    refute sop_other.can_manage?(person_to_keep)
+    refute sop_keep.can_manage?(other_person)
+    assert sop_both.can_manage?(other_person)
+    assert sop_other.can_manage?(other_person)
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+    sop_keep.reload
+    sop_both.reload
+    sop_other.reload
+    sop_keep.permission_for = nil
+    sop_both.permission_for = nil
+    sop_other.permission_for = nil
+
+    assert sop_keep.can_manage?(person_to_keep)
+    assert sop_both.can_manage?(person_to_keep)
+    assert sop_other.can_manage?(person_to_keep)
+    assert 1, sop_both.policy.permissions.length
+    assert 3, Permission.where(contributor_type: 'Person', contributor_id: person_to_keep.id).length
```

Comment:
would be good to include cases as well as MANAGE and check the final values on the permissions

Reply:
Resolved in [122eb02](https://github.com/seek4science/seek/pull/1683/commits/122eb0220d3305143a3492c92d1e0352ff639a6c).